lib/src/test/java/com/basistheory/android/service/HTTPClientTests.kt

```diff
@@ -0,0 +1,200 @@
+import junitparams.JUnitParamsRunner
+import kotlinx.coroutines.Dispatchers
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import kotlinx.coroutines.runBlocking
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertThrows
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import java.io.IOException
+
+@RunWith(JUnitParamsRunner::class)
+class HttpClientTests {
+    private lateinit var server: MockWebServer
+    private lateinit var client: HttpClient
+
+
+    private val endpoint = "/api"
+
+    @Before
+    fun setUp() {
+        server = MockWebServer()
+        server.start()
+
+        client = HttpClient(Dispatchers.IO)
+    }
+
+    @After
+    fun tearDown() {
+        server.shutdown()
+    }
+
+    @Test
+    fun testGetRequest() = runBlocking {
+        server.enqueue(MockResponse().setBody("Got it"))
+
+        val url = server.url(endpoint).toString()
+
+        client.get(url)
+
+        val request = server.takeRequest()
+        assertEquals("GET", request.method)
+        assertEquals(endpoint, request.path)
+    }
+
+    @Test
+    fun testPostRequest() = runBlocking {
+        val expectedRequestBody = "Request body"
+        server.enqueue(MockResponse().setBody("Posted it"))
+
+        val url = server.url(endpoint).toString()
+
+        client.post(url, expectedRequestBody)
+
+        val request = server.takeRequest()
+        assertEquals("POST", request.method)
+        assertEquals(endpoint, request.path)
+        assertEquals(expectedRequestBody, request.body.readUtf8())
```

Comment:
can you assert on the response of all of these?

Reply:
Yes, I can. Do you see any value tho? Or is there anything in particular you'd like to see? Because it's pretty much the same as above

Reply:
in which one are you asserting the response?

Reply:
just to make sure nothing is updating the response and we get what we expect

Reply:
i'll leave it up to you